### PR TITLE
dockerfile: create stub file with frontmatter

### PR DESCRIPTION
**Pull request by crazy-max**
needs https://github.com/moby/buildkit/pull/2970

as discussed, it's better to have the frontmatter section here instead of the buildkit repo. ~needs to fetch remote reference in the same folder otherwise we can't include it in the stub file. will be fixed anyway when we are going to move to the new location in https://github.com/docker/docker.github.io/pull/15110~

### CODE REVIEW

**Comment by crazy-max** [comment]
actually not so great atm as we have to fetch remote reference in the same folder so we can include it with liquid template func `include` or `include_file`. problem is this file could be discovered by the end user. keeping in draft for now, we need to find a better way.

File: engine/reference/builder.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+title: Dockerfile reference
+description: "Dockerfiles use a simple DSL which allows you to automate the steps you would normally manually take to create an image."
+keywords: build, dockerfile, reference
+toc_max: 3
+redirect_from:
+- /reference/builder/
+---
+
+{% github_content site.buildkit.repo site.buildkit.ref site.buildkit.default_branch "frontend/dockerfile/docs/reference.md" %}
```
Comment (crazy-max) [inline engine/reference/builder.md:10]: ok found a nice way to inject remote content using a custom plugin implementing a liquid tag. Let me know what you think about it @thaJeztah

Comment (crazy-max) [reply engine/reference/builder.md:10]: Here are the logs:

```
7:46:43 AM: Starting plugin github_content.rb...
7:46:43 AM:   Inject https://raw.githubusercontent.com/crazy-max/buildkit/docs-ref-move-frontmatter/frontend/dockerfile/docs/reference.md to engine/reference/builder.md
7:46:43 AM:     edit_url:  https://github.com/crazy-max/buildkit/edit/master/frontend/dockerfile/docs/reference.md
7:46:43 AM:     issue_url: https://github.com/crazy-max/buildkit/issues/new?body=File: [frontend/dockerfile/docs/reference.md](https://docs.docker.com/engine/reference/builder.md)
7:46:43 AM: done in 0.105188682 seconds
```

File: _config.yml [hunk 2]
```diff
@@ -47,6 +47,11 @@ compose_file_v2: "2.4"
 machine_version: "0.16.0"
 distribution_version: "2.7"
 compose_switch_version: "1.0.4"
+buildkit:
+  repo: crazy-max/buildkit
+  ref: docs-ref-move-frontmatter
```
Comment (crazy-max) [inline _config.yml:52]: use my fork for testing

**Comment by thaJeztah** [comment]
> actually not so great atm as we have to fetch remote reference in the same folder so we can include it with liquid template func include or include_relative. 

I'm a bit confused though; for other includes, we use the `_includes` directory (and subdirectories thereof) can't we do the same for this?

**Comment by thaJeztah** [comment]
Or is that because of relative links to markdown files?

**Comment by crazy-max** [comment]
> I'm a bit confused though; for other includes, we use the `_includes` directory (and subdirectories thereof) can't we do the same for this?

Hum indeed it might work with the `_inlcudes` folder as it's not rendered :+1:

File: engine/reference/builder.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+title: Dockerfile reference
+description: "Dockerfiles use a simple DSL which allows you to automate the steps you would normally manually take to create an image."
+keywords: build, dockerfile, reference
+toc_max: 3
+redirect_from:
+- /reference/builder/
+---
+
+{% github_content site.buildkit.repo site.buildkit.ref site.buildkit.default_branch "frontend/dockerfile/docs/reference.md" 2 -1 %}
```
Comment (jedevc) [inline engine/reference/builder.md:10]:
```suggestion
{% github_content site.buildkit.repo site.buildkit.ref site.buildkit.default_branch "frontend/dockerfile/docs/reference.md" 1 -1 %}
```
I think the whitespace in front will get dropped by jekyll? Then this just becomes "skip the first line" which seems clearer to me.

File: _config.yml [hunk 4]
```diff
@@ -133,52 +133,54 @@ fetch-remote:
...
-    ref: "master"
+    ref: "docs-ref-move-frontmatter"
     paths:
-      - dest: "engine/reference/builder.md"
+      - dest: "_includes/dockerfile/reference.md"
         src:
-          - "frontend/dockerfile/docs/reference.md"
+          - file: "frontend/dockerfile/docs/reference.md"
+            line_start: 2
+            line_end: -1
```
Comment (crazy-max) [inline _config.yml:186]: @thaJeztah As discussed, removed the `github_content` liquid tag plugin and enhance our `fetch_remote` plugin to be able to specify `line_start` and  `line_end`

File: engine/reference/builder.md (hunk 3 above)
Comment (crazy-max) [reply engine/reference/builder.md]: Should be ok as there is an extra line after the frontmatter section

File: _config.yml (hunk 4 above)
Comment (crazy-max) [reply _config.yml:186]: Argh `edit_url` and `issue_url` is lost then :persevere:

Comment (crazy-max) [reply _config.yml]: Ok managed to fix `edit_url` and `issue_url` inclusion by grabbing the one available in site default scope. I have created a dedicated liquid tag named `include_remote` that does pretty much the same job as `include` except it also handles `<line_start>`, `<line_end>` and creates the right `edit_url` and `issue_url`.

LGTY @thaJeztah ?

File: _config.yml [hunk 5]
```diff
@@ -175,10 +175,10 @@ fetch-remote:
         src:
           - "docs/configuration.md"
 
-  - repo: "https://github.com/moby/buildkit"
+  - repo: "https://github.com/crazy-max/buildkit"
     default_branch: "master"
-    ref: "master"
+    ref: "docs-ref-move-frontmatter"
```
Comment (crazy-max) [inline _config.yml:180]: use my fork while waiting for https://github.com/moby/buildkit/pull/2970 to be merged

**Review by thaJeztah — APPROVED** [review]
LGTM after switching to upstream once https://github.com/moby/buildkit/pull/2970 is merged

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**